### DISCUSSION TITLE
“Open” Boundary Condition for Species Fluxes?

### DISCUSSION THREAD

**Discussion by ghost** [idaholab/moose#20533] [Q&A General] [answered]
My goal is to have a defined flux of species and associated energy coming into a chamber, reacting inside, and then leaving freely through an outlet on the other end. Setting a FunctionNeumann on the inlet and proper BCs where there is a solid surface is easy enough, but the "open" boundary at the outlet appears to be much trickier.

The system I'm simulating primarily uses the application Zapdos/Crane, but I was wondering if there exists a BC in the basic MOOSE framework or in one of its modules that computes the values of a given variable on the boundary to be the same values in the elements/points adjacent to it, or perhaps something similar to this to effectively "open" the boundary?

Edit: I came across "PorousFlowOutflowBC" which seems to do exactly what I want from the description of it, but I've never used this module before and I'm unsure of what I would need to add to my simulation to make it work, if anything. "ChemicalOutFlowBC" also seems like an option, but I'm confused about the porosity term.

Thank you!

**Comment by GiudGiud** [comment, marked as answer]
Hello

Is this a flow simulation? If so then you could set a pressure boundary condition for an outlet. 
Is there an advective component and/or a diffusive component?

Guillaume

**Reply by ghost** [reply]
Well, it's a weird mix because I'm simulating the plasma using Zapdos/Crane on top of a background flow from INS. I ran a simple INS input file with a defined velocity at the inlet, no slip everywhere else, and left the outlet undefined and it produced a velocity field as expected. I used this INS exodus file as the mesh for my plasma simulation, and I have a kernel that couples my plasma species to the flow field for a "convection" (not sure if thats the correct term; just going off what the kernel is called). My thought process for this is that the background gas is much higher in density than the plasma species, so any plasma interactions have very little effect on the flow field. The plasma species indeed experience diffusion and advection from the electric fields. (I'm unsure if this advection is the same as what you're thinking; It seems advection & convection are often used interchangeably and mean different things in different contexts.) 

I hope this helped clarify!

**Reply by GiudGiud** [reply]
So a BC with values at the boundary being the same as values on the element would be 0 normal face gradient or no flux. 

In flow this is typically a fully developed conditions, where while there is a pressure gradient, the velocity profile does not change when those are reached. In the INS-FE part of the NS module, I think this is https://mooseframework.inl.gov/source/bcs/INSMomentumNoBCBCLaplaceForm.html
or a Dirichlet boundary condition on pressure. 

@lindsayad knows more about that part of the NS module

**Reply by lindsayad** [reply]
If your advection term is not integrated by parts, then a natural boundary condition (no bc specification at all) will often work well. However, this does implicitly impose a zero-gradient condition as @GiudGiud stated since you are implicitly saying that your diffusive flux is zero. If you want no conditions on the boundary, then you have to code up BCs of the variety that @GiudGiud linked-to. Essentially you will want to write in code-form the surface terms that result when you integrate your weak form by parts. If you are working with Zapdos, I see there are a lot of `*DoNothingBC` classes that I wrote a very long time ago. I can't guarantee that they still work (they are probably mostly untested) but the concepts in there should be good
